src/HospitalLibrary/Patients/PatientService.cs

```
@@ -23,19 +24,15 @@ public Task<IEnumerable<Patient>> GetAll()
         {
             return _unitOfWork.PatientRepository.GetAll();
         }
-        
         public async Task<Patient> Create(Patient patient)
         {
-            try
-            {
-                _unitOfWork.PatientRepository.Add(patient);
-                _unitOfWork.PatientRepository.Save();
-            }
-            catch(Exception  ex)
-            {
-                throw new BadRequestException("Uid or Email is already taken");
-            }
-            return patient;
+            if (_unitOfWork.PersonRepository.GetOneByEmail(patient.Email) != null)
```

Let the database throw exception and throw yours after